src/d3a/models/strategy/predefined_pv.py

```diff
@@ -92,6 +93,10 @@ def set_produced_energy_forecast_kWh_future_markets(self, reconfigure=True):
             self._read_predefined_profile_for_pv()
         for market in self.area.all_markets:
             slot_time = market.time_slot
+            if not self.power_profile:
+                raise D3AException(
+                    f"PV {self.owner.name} tries to set its energy forecast without a "
+                    f"power profile.")
```

Comment:
Hmm, i wonder when this case can happen. Only when `read_arbitrary_profile` returns None, in line 124, right?

Reply:
Either when it returns None or empty dict, yes. Also, if `area_reconfigure_event` is called with an empty dict, which was the case that we saw in the traceback.

Reply:
I see, then I propose to add this check in the predefined load, too:
https://github.com/gridsingularity/d3a/blob/e49b04bfcb4ddb73782b8defdb4598a8ee0d8f10/src/d3a/models/strategy/predefined_load.py#L108

Reply:
Agreed, adding it